demos/adjoint/adjoint_demo.py

```diff
@@ -0,0 +1,348 @@
+# Adjoint Inverse Demo
+# ================================
+#
+# Introduction
+# ------------
+# In this tutorial, we will demonstrate how to perform an inversion to recover the initial temperature field of an idealised mantle convection simulation using G-ADOPT.
```

Comment:
I would flag here that the total runtime for this tutorial is XX hours. And a user should have a cup of tea ready!

Reply:
I am now thinking to make this tutorial for only 5 timesteps, and explain to the user that for replicating the study in Ghelichkhan et al you should change the `initial_timestep` time-step to 0 instead of `timesteps - 5`.